src/Microsoft.Identity.Abstractions/DownstreamRestApi/IDownstreamRestApi.cs

```diff
@@ -12,74 +12,81 @@ namespace Microsoft.Identity.Abstractions
     /// <summary>
     /// Interface used to call a downstream REST API, for instance from controllers.
     /// </summary>
-    public interface IDownstreamRestApi
+    public partial interface IDownstreamRestApi
     {
         /// <summary>
-        /// Calls the downstream REST API on behalf of the user, based on a description of the
-        /// downstream REST API in the configuration (service name).
+        /// Calls the downstream REST API based on a description of the
+        /// downstream REST API in the configuration (service name), overridatable programmatically. The choice
+        /// of calling the API on behalf of the user or the app, is made by the configuration or programmatically. 
+        /// This is the lowest level API. There are other APIs for specific Http methods.
         /// </summary>
         /// <param name="serviceName">Name of the service describing the downstream web API. There can
         /// be several configuration named sections mapped to a <see cref="DownstreamRestApiOptions"/>,
-        /// each for one downstream REST API. You can pass-in null, but in that case <paramref name="calledDownstreamRestApiOptionsOverride"/>
+        /// each for one downstream REST API. You can pass-in null, but in that case <paramref name="downstreamRestApiOptionsOverride"/>
         /// needs to be set.</param>
-        /// <param name="calledDownstreamRestApiOptionsOverride">(Optional) Overrides the options proposed in the configuration described
-        /// by <paramref name="serviceName"/>.</param>
+        /// <param name="downstreamRestApiOptionsOverride">(Optional) Overrides the options proposed in the configuration described
+        /// by <paramref name="serviceName"/>. Can be null, in which case only the configuration is applied.</param>
         /// <param name="user">(Optional) Claims representing a user. This is useful on platforms like Blazor
         /// or Azure Signal R, where the HttpContext is not available. In other platforms, the library
         /// will find the user from the HTTP request context.</param>
         /// <param name="content">Content to send to the REST API in the case where
-        /// <see cref="DownstreamRestApiOptions.HttpMethod"/> is <code>HttpMethod.Patch</code>, 
+        /// <see cref="AuthorizationHeaderProviderOptions.HttpMethod"/> is <code>HttpMethod.Patch</code>, 
         /// <see cref="HttpMethod.Post"/>, <see cref="HttpMethod.Put"/>.</param>
         /// <param name="cancellationToken">Cancellation token.</param>
         /// <returns>An <see cref="HttpResponseMessage"/> that the application will process.</returns>
-        Task<HttpResponseMessage> CallRestApiForUserAsync(
+        public Task<HttpResponseMessage> CallRestApiAsync(
```

Comment:
I am thinking the API: CallRestApiAsync(DownstreamRestApiOptions, ClaimsPrincipal, HttpContext, CancellationToken) would be very simple to do.

Reply:
Right, I was wondering about bringing it. It does exist in the implementation.

Reply:
Added